benchmark/scripts/compare_perf_tests.py

```diff
@@ -18,40 +18,270 @@
 import argparse
 import csv
 import sys
-
-TESTNAME = 1
-SAMPLES = 2
-MIN = 3
-MAX = 4
-MEAN = 5
-SD = 6
-MEDIAN = 7
-
-HTML = """
+from math import sqrt
+
+
+class PerformanceTestResult:
+    def __init__(self, csv_row):
+        # csv_row[0] is just an ordinal number of the test - skip that
+        self.name = csv_row[1]          # Name of the performance test
+        self.samples = int(csv_row[2])  # Number of measurement samples taken
+        self.min = int(csv_row[3])      # Minimum runtime (ms)
+        self.max = int(csv_row[4])      # Maximum runtime (ms)
+        self.mean = int(csv_row[5])     # Mean (average) runtime (ms)
+        sd = int(csv_row[6])            # Standard Deviation (ms)
+        # For computing running variance
+        self.S_runtime = (0 if self.samples < 2 else
+                          (sd * sd) * (self.samples - 1))
+        self.median = int(csv_row[7])   # Median runtime (ms)
+        self.max_rss = int(csv_row[8])  # Maximum Resident Set Size (B)
```

Comment:
This line also breaks when I ask this script to compare benchmark executable output, as there are only 8 columns there.